tensornetwork/tensor.py

```diff
@@ -185,3 +185,18 @@ def __matmul__(self, other: "Tensor") -> "Tensor":
       raise ValueError(errstr)
     array = self.backend.matmul(self.array, other.array)
     return Tensor(array, backend=self.backend)
+
+  def __call__(self, *args):
+    return NconBuilder([self], [list(args)]) 
+
+
+class NconBuilder():
+  def __init__(self, tensors, axes):
+    self.tensors = tensors[:] # Forces a copy.
```

Comment:
wait, why do we want a copy?

Reply:
Doesn't matter too much tbh. Just guarantees the function to be pure.

Reply:
I see. I suppose that is important, but it could have significant overhead. Let's leave it though yes, I think there's no other way.